sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/literals.scala

```diff
@@ -265,6 +267,19 @@ object Literal {
       s"Literal must have a corresponding value to ${dataType.catalogString}, " +
       s"but class ${Utils.getSimpleName(value.getClass)} found.")
   }
+
+  def fromSQL(sql: String): Expression = {
+    CatalystSqlParser.parseExpression(sql).transformUp {
```

Comment:
Looking at the test failure, the missing piece is that `Literal#sql` can generate CAST which needs to be resolved to attach timezone. So we should add one more case match
```
case c: Cast if c.needsTimeZone =>
  c.withTimeZone(conf.sessionLocalTimeZone)
```

Reply:
Thanks for the pointer, saved me a lot of time!